### ISSUE TITLE
Infinite publishing loop

### ISSUE THREAD

**Issue by nealalpert** [contentful/contentful-import#49]
While attempting to import a space that has been exported with the contentful-export tool we're running into an infinite loop.

<img width="544" alt="screen shot 2017-06-01 at 12 35 51 pm" src="https://cloud.githubusercontent.com/assets/2166663/26692597/e7f99b6e-46c6-11e7-8770-39cf2d5b662b.png">

The item in question and its children all seem to be in the published state so we're fairly perplexed.

```
{
            "sys": {
                "space": {
                    "sys": {
                        "type": "Link",
                        "linkType": "Space",
                        "id": "ists1rc8crak"
                    }
                },
                "id": "6gtif6Mpk4OoMUSGsyoSQg",
                "type": "Entry",
                "createdAt": "2017-03-22T02:14:09.656Z",
                "updatedAt": "2017-05-08T03:34:50.081Z",
                "createdBy": {
                    "sys": {
                        "type": "Link",
                        "linkType": "User",
                        "id": "3gbtnWxsCVbzkTwBr1xBeI"
                    }
                },
                "updatedBy": {
                    "sys": {
                        "type": "Link",
                        "linkType": "User",
                        "id": "5V3LRH02gkkklSapakIetU"
                    }
                },
                "publishedCounter": 9,
                "version": 127,
                "publishedBy": {
                    "sys": {
                        "type": "Link",
                        "linkType": "User",
                        "id": "5V3LRH02gkkklSapakIetU"
                    }
                },
                "publishedVersion": 126,
                "firstPublishedAt": "2017-03-22T02:15:28.310Z",
                "publishedAt": "2017-05-08T03:34:50.031Z",
                "contentType": {
                    "sys": {
                        "type": "Link",
                        "linkType": "ContentType",
                        "id": "globalContent"
                    }
                }
            },
            "fields": {
                "name": {
                    "en-US": "Title"
                },
                "title": {
                    "en-US": "Title"
                },
                "description": {
                    "en-US": "description statement"
                },
                "logo": {
                    "en-US": {
                        "sys": {
                            "type": "Link",
                            "linkType": "Entry",
                            "id": "5pRQJqbHy0wMiyGoQuscYg"
                        }
                    }
                },
                "globalNavigation": {
                    "en-US": [
                        {
                            "sys": {
                                "type": "Link",
                                "linkType": "Entry",
                                "id": "2mEIfdMKVa0AU8wQ82COUc"
                            }
                        },
                        {
                            "sys": {
                                "type": "Link",
                                "linkType": "Entry",
                                "id": "6Ghbmtw80g8Mo42QQIcoeE"
                            }
                        }
                    ]
                },
                "openText": {
                    "en-US": "open menu"
                },
                "mobileOpenIcon": {
                    "en-US": {
                        "sys": {
                            "type": "Link",
                            "linkType": "Asset",
                            "id": "71ARQAh8vC4ca4Oam8Mw0Y"
                        }
                    }
                },
                "closeText": {
                    "en-US": "close menu"
                },
                "mobileCloseIcon": {
                    "en-US": {
                        "sys": {
                            "type": "Link",
                            "linkType": "Asset",
                            "id": "2cbjifRK2YOge2eICAc4Ig"
                        }
                    }
                },
                "socialNavigation": {
                    "en-US": [
                        {
                            "sys": {
                                "type": "Link",
                                "linkType": "Entry",
                                "id": "2yPd9sD4WsYMOIsa8kYWko"
                            }
                        }
                    ]
                },
                "copyright": {
                    "en-US": "All rights reserved.  All trademarks are the property of their respective owners."
                },
                "utilityNavigation": {
                    "en-US": [
                        {
                            "sys": {
                                "type": "Link",
                                "linkType": "Entry",
                                "id": "1ynHPLKZmY6aksEMeq4UQu"
                            }
                        },
                        {
                            "sys": {
                                "type": "Link",
                                "linkType": "Entry",
                                "id": "4Lmxm4yhS0g64YmIEsUwsm"
                            }
                        },
                        {
                            "sys": {
                                "type": "Link",
                                "linkType": "Entry",
                                "id": "67LAQ7kMpOIUQKgsmygEgU"
                            }
                        }
                    ]
                },
                "socialSiteTitle": {
                    "en-US": "Title"
                },
                "socialPageTitle": {
                    "en-US": "Welcome"
                },
                "socialDescription": {
                    "en-US": "description statement"
                },
                "socialUrl": {
                    "en-US": "url"
                },
                "socialImage": {
                    "en-US": {
                        "sys": {
                            "type": "Link",
                            "linkType": "Asset",
                            "id": "5awubBLrri0cQSssgoO04"
                        }
                    }
                },
                "socialImageAlt": {
                    "en-US": "Save"
                },
                "twitterCardStyle": {
                    "en-US": "summary_large_image"
                },
                "twitterHandle": {
                    "en-US": "@twitter"
                }
            }
        },
```

**Comment by eutopian** [comment]
I'm getting the same error... any luck?

**Comment by nealalpert** [comment]
Yes and no.  I was unable to determine why the export in question was failing to import.  It seemed to stall on one of the global navigation links that were part of the global header/footer object that I posted.

I re-exported the space with the option not to include draft content (according to the documentation on contentful-export the default export behavior excludes draft content so this extra option probably did nothing) and it imported correctly this time.

**Comment by eutopian** [comment]
Thanks for the suggestion... I restarted the import, seems to be running okay at the moment, but if it runs into the loop again I'll exclude the drafts manually as well (or maybe you actually just re-executed the task like me... I actually suspect it's a rate limit thing)

**Comment by eutopian** [comment]
Anyhow... an update... I figured out what the problem was for me tho... some of the contents that were previously published in one space, now have added required fields that are blank. These contents can't get published when I import, and anything they're linked to that requires them to be published could also not get published.

**Comment by nealalpert** [comment]
I assumed the rate limit messages were a kind of throttling to prevent DDoS attacks on Contentful servers but that's just speculation on my part.

We did have some published content that had a required field added to the content model after they were published but it didn't seem to cause the infinite loop.  It did cause the import script to be unable to publish them after they were imported though.  I suspect that that is expected behavior.  We cleaned it up anyway just in case.
